types/markdown-it/lib/parser_block.d.mts

```diff
@@ -3,15 +3,13 @@ import Ruler from "./ruler.mjs";
 import StateBlock from "./rules_block/state_block.mjs";
 import Token from "./token.mjs";
 
-declare namespace ParserBlock {
-    type RuleBlock = (state: StateBlock, startLine: number, endLine: number, silent: boolean) => boolean;
-}
+export type RuleBlock = (state: StateBlock, startLine: number, endLine: number, silent: boolean) => boolean;
```

Comment:
can you use something like:
```ts
declare type RuleBlockFn = (state: StateBlock, startLine: number, endLine: number, silent: boolean) => boolean;
export type RuleBlock = RuleBlockFn;

declare namespace ParserBlock {
    /**
     * @deprecated
     * Use:
     * ```js
     * import { RuleBlock } from "markdown-it/lib/parser_block.mjs"`
     * ```
     */
    type RuleBlock = RuleBlockFn;
}
```
that will not break existing code (still at 14.0). Not sure what is direct use-case requiring this rewrite

Reply:
@peterblazejewicz  It was my mistake, I believe. I migrated the typedefs for this lib over to ES modules last week. As I did that I moved away from namespaces to directly exported symbols. I did it with all the other modules, but must have simply forgot this one.

Reply:
> @peterblazejewicz  It was my mistake, I believe. I migrated the typedefs for this lib over to ES modules last week. As I did that I moved away from namespaces to directly exported symbols. I did it with all the other modules, but must have simply forgot this one.

yes, this the point. only this is forgotten to be migrate.

namespace should be completely being removed in v14

Reply:
Yes, I would agree this is a bug. My code dependant project only uses inline rules, so this particular module was never tested on my own code when I made the migration:

If I modify my code a bit the error is glaring:

```js
/**
 * @typedef {import("markdown-it/lib/parser_inline.mjs").RuleInline} RuleInline
 * @typedef {import("markdown-it/lib/parser_block.mjs").RuleBlock} RuleBlock
 * @typedef {import("markdown-it/lib/rules_inline/state_inline.mjs").default} StateInline
 */
```

`RuleInline` works fine, however `RuleBlock` fails to typecheck:

```
lib/ruler.js:3:57 - error TS2694: Namespace '"/my-lib/node_modules/@types/markdown-it/lib/parser_block"' has no exported member 'RuleBlock'.

3  * @typedef {import("markdown-it/lib/parser_block.mjs").RuleBlock} RuleBlock
```